### ISSUE TITLE
Autotune runs more than needed

### ISSUE THREAD

**Issue by claudioandre-br** [magnumripper/JohnTheRipper#3516]
In the example below it runs 3 times.

```
$ rm ../run/*.pot; LWS=128 ../run/john -form=nt-opencl -mask:?l -min-len=7 -max-len=9 alltests.in -dev=2 --verb=5 --skip
```

How `reset()` is working now?

**Comment by claudioandre-br** [comment]
So, this is a bug that seems to be in many GPU mask enabled formats. It is very low priority.
****

BTW: is this warning correct? If I have the necessary minimum, why was I warned?
```
Warning: Only 172288 candidates buffered, minimum 172288 needed for performance.
```

**Comment by magnumripper** [comment]
I started to work on this problem in general a week ago or two but much more interesting work got my attention.

**Comment by magnumripper** [comment]
FWIW that erroneous warning was fixed in b30c280eb

**Comment by magnumripper** [comment]
This will automagically be revisited when we fix #3249 and #3697 later

**Comment by magnumripper** [comment]
Closing in favor of #4012
